### PR TITLE
Move zabbix state tests to pytest

### PR DESCRIPTION
**Pull request by krionbsd**
### What does this PR do?
Move zabbix state tests to pytest

### CODE REVIEW

File: tests/pytests/unit/states/test_zabbix_action.py [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+            ],
+        },
+        "eventsource": "2",
+        "name": "Auto registration Databases",
+        "operations": [{"opgroup": [{"groupid": "6"}], "operationtype": "4"}],
+    }
+
+
+@pytest.fixture
+def exising_obj():
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_zabbix_action.py:27]:
```suggestion
def existing_obj():
```

File: tests/pytests/unit/states/test_zabbix_template.py [hunk 2]
```diff
@@ -0,0 +1,331 @@
+"""
+    :codeauthor: :email:`Jakub Sliva <jakub.sliva@ultimum.io>`
+"""
+
+import pytest
+import salt.states.zabbix_template as zabbix_template
+from tests.support.mock import MagicMock, patch
+
+# INPUT_PARAMS = {"applications": [{"name": "Ceph OSD"}]}
+
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_zabbix_template.py:10]:
```suggestion
```

File: tests/pytests/unit/states/test_zabbix_template.py [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+            Differentiate between __salt__ exec module function calls with different parameters.
+            """
+            if args[0] == "template.get":
+                return []
+            elif args[0] == "template.create":
+                return {"templateids": ["10206"]}
+            elif args[0] == "application.get":
+                return []
+            elif args[0] == "application.create":
+                return {"applicationids": ["701"]}
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_zabbix_template.py:163]:
```suggestion
            if args[0] in ("template.get", "application.get"):
                return []
            elif args[0] == "template.create":
                return {"templateids": ["10206"]}
            elif args[0] == "application.create":
                return {"applicationids": ["701"]}
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: tests/pytests/unit/states/test_zabbix_action.py (hunk 1 above)
Comment (krionbsd) [reply tests/pytests/unit/states/test_zabbix_action.py]: Fixed

File: tests/pytests/unit/states/test_zabbix_template.py (hunk 2 above)
Comment (krionbsd) [reply tests/pytests/unit/states/test_zabbix_template.py]: Fixed

File: tests/pytests/unit/states/test_zabbix_template.py (hunk 3 above)
Comment (krionbsd) [reply tests/pytests/unit/states/test_zabbix_template.py]: Fixed

File: tests/pytests/unit/states/test_zabbix_template.py [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+        def side_effect_run_query(*args):
+            """
+            Differentiate between __salt__ exec module function calls with different parameters.
+            """
+            if args[0] in ("template.get", "application.get"):
+                return []
+            elif args[0] == "template.create":
+                return {"templateids": ["10206"]}
+            elif args[0] == "application.get":
+                return []
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_zabbix_template.py:159]:
```suggestion
```
We are now covering this branch in line 154

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: tests/pytests/unit/states/test_zabbix_template.py (hunk 4 above)
Comment (krionbsd) [reply tests/pytests/unit/states/test_zabbix_template.py]: Fixed

**Review by s0undt3ch — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**